src/components/execution/AddExecutionForm.js

```diff
@@ -237,23 +277,76 @@ class AddExecutionForm extends Component {
     );
   };
 
+  renderAlgorithmsAndPipelinesSelect = () => {
+    const { algorithmId, pipelineId } = this.state;
+    const { classes, t, algorithms, pipelines } = this.props;
+    const algorithmMenuItems = algorithms.map(({ id, name }) => (
+      <MenuItem
+        value={id}
+        key={id}
+        className={classes.menuItem}
+        id={buildExecutionAlgorithmOptionId(id)}
+      >
+        {name}
+      </MenuItem>
+    ));
+
+    const pipelineMenuItems = pipelines.map(({ id, name }) => (
+      <MenuItem
+        value={id}
+        key={id}
+        className={classes.menuItem}
+        id={buildExecutionPipelineOptionId(id)}
+      >
+        {name}
+      </MenuItem>
+    ));
+
+    const algorithmLabel = `${t('Algorithm')} ${t('(Required)')}`;
+
+    return (
+      <FormControl variant="outlined" className={classes.formControl}>
+        <InputLabel id="algorithm-select">{algorithmLabel}</InputLabel>
+        <Select
+          labelId="algorithm-select"
+          value={pipelineId !== '' ? pipelineId : algorithmId}
+          onChange={this.handleAlgorithmSelectOnChange}
+          label={algorithmLabel}
+          id={EXECUTIONS_ALGORITHMS_SELECT_ID}
+        >
+          {!algorithmMenuItems.isEmpty() && (
+            <ListSubheader>{t('Algorithms')}</ListSubheader>
+          )}
+          {algorithmMenuItems}
+          {!pipelineMenuItems.isEmpty() && (
+            <ListSubheader>{t('Pipelines')}</ListSubheader>
+          )}
+          {pipelineMenuItems}
+        </Select>
+      </FormControl>
+    );
+  };
+
   renderExecuteButton = () => {
-    const { sourceId, algorithmId, parameters } = this.state;
-    const { t, pythonVersion, classes } = this.props;
+    const { sourceId, algorithmId, parameters, pipelineId } = this.state;
+    const { t, pythonVersion, classes, pipelines } = this.props;
 
     const valid =
       sourceId &&
       algorithmId &&
       pythonVersion?.valid &&
       (!parameters || areParametersValid(parameters));
+    const validPipeline = sourceId && pipelines;
     const button = (
       <div className={classes.buttonWrapper}>
         <Button
           id={EXECUTIONS_EXECUTE_BUTTON_ID}
           variant="contained"
           color="primary"
-          onClick={this.executeAlgorithm}
-          disabled={!valid}
+          onClick={
+            pipelineId !== '' ? this.executePipeline : this.executeAlgorithm
```

Comment:
I think that adding an execution type doesn't add that much clarity to the code since their's only a few places where we check for the `pipelineId` condition.